lib/Bot/BasicBot/Pluggable/Module/Join.pm

```diff
@@ -9,7 +9,7 @@ sub connected {
 
     ## If we are not a array reference, we are problably the old
     ## string format ... trying to convert
-    if ( not ref($channels) eq 'ARRAY' ) {
+    if ( not ref($channels) && $channels =~ 'ARRAY' ) {
```

Comment:
This fix is for DBM::Deep and similar backing stores; the channels are an object of DBM::Deep::Array which can be derived as array, but the "ref" function returns the class and not the storage type in that case